Lib/fontbakery/profiles/universal.py

```diff
@@ -2212,6 +2213,189 @@ def com_google_fonts_check_math_signs_width(ttFont):
         yield PASS, "Looks good."
 
 
+@check(
+    id="com.google.fonts/check/tabular_kerning",
+    rationale="""
+        Tabular glyphs should not have kerning, as they are meant to be used in tables.
+
+        This check looks for kerning in:
+        - all glyphs in a font in combination with tabular numerals;
+        - tabular symbols in combination with tabular numerals.
+
+        "Tabular symbols" is defined as:
+        - for fonts with a "tnum" feature, all "tnum" substitution target glyphs;
+        - for fonts without a "tnum" feature, all glyphs that have the same width
+        as the tabular numerals, but limited to numbers, math and currency symbols.
+
+        This check may produce false positives for fonts with no "tnum" feature
+        and with equal-width numerals (and other same-width symbols) that are
+        not intended to be used as tabular numerals.
+    """,
+    proposal="https://github.com/fonttools/fontbakery/issues/4440",
+)
+def com_google_fonts_check_tabular_kerning(ttFont):
+    """Check tabular widths don't have kerning."""
+    from vharfbuzz import Vharfbuzz
+    import uharfbuzz as hb
```

Comment:
Why do you need both vharfbuzz and uharfbuzz, vharfbuzz is a nicer wrapper around uharfbuzz?

Reply:
I'm using uharfbuzz to construct a buffer made of glyphs rather than characters which vharfbuzz doesn't expose, to measure kerning between arbitrary glyph combinations.